cli.js

```diff
@@ -181,29 +187,45 @@ if (require.main === module) {
                 type: 'string',
                 required: 'true',
                 default: hecate.port
-            }];
-
-            // if the username and password hasn't already been set, prompt for it
-            if (!hecate.user) {
-                args = args.concat(auth(hecate.user));
-            }
-
-            prompt.get(args, (err, res) => {
+            }], (err, res) => {
                 if (err) throw err;
-
                 hecate.url = res.url;
                 hecate.port = res.port;
-
-                if (res.hecate_username && res.hecate_password) {
-                    hecate.user = {
-                        username: res.hecate_username,
-                        password: res.hecate_password
-                    };
-                }
-
                 argv.cli = true;
 
-                return run();
+                // if a custom auth policy hasn't been passed
+                if (!hecate.auth_rules) {
+                    // fetch auth
+                    hecate.auth({}, (err, auth_rules) => {
+                        // if requesting auth returns a 401
+                        if (err.message === '401: Unauthorized') {
+                            // if username and password isn't set, prompt for it
+                            if (!hecate.user) {
+                                prompt.get(auth(hecate.user), (err, res) => {
+                                    if (err) throw err;
+                                    hecate.user = {
+                                        username: res.hecate_username,
+                                        password: res.hecate_password
+                                    };
+                                    // request auth again
+                                    hecate.auth({}, (err, auth_rules) => {
+                                        if (err) throw err;
+                                        hecate.auth_rules = auth_rules;
+                                        return run();
+                                    });
+                                });
+                            } else {
+                                console.error();
```

Comment:
It's possible for the user not to have access to the auth endpoint, but to still have access to the endpoint that they are trying to request. Let's just pass through to `run()` here.